nilearn/glm/first_level/first_level.py

```diff
@@ -124,26 +145,45 @@ def run_glm(Y, X, noise_model='ar1', bins=100, n_jobs=1, verbose=0):
     # Create the model
     ols_result = OLSModel(X).fit(Y)
 
-    if noise_model == 'ar1':
-        # compute and discretize the AR1 coefs
-        ar1 = (
-            (ols_result.residuals[1:]
-             * ols_result.residuals[:-1]).sum(axis=0)
-            / (ols_result.residuals ** 2).sum(axis=0)
-        )
+    if 'ar' in noise_model:
+        ar_order = int(re.split('ar', noise_model)[1])
```

Comment:
I would add a little bit more checking here. For example, if the (distracted) user provides `arfoo` or `arr3` this would fail with a non informative error.

Reply:
I tried to catch a few more edge cases. Please let me know if you think of others I missed.